internal/datacoord/garbage_collector.go

```diff
@@ -566,3 +566,61 @@ func (gc *garbageCollector) recycleUnusedIndexFiles() {
 			zap.Int("delete index files num", deletedFilesNum))
 	}
 }
+
+func (gc *garbageCollector) recycleUnusedAnalyzeFiles() {
+	log.Info("start recycleUnusedAnalyzeFiles")
+	ctx, cancel := context.WithCancel(context.Background())
+	defer cancel()
+	startTs := time.Now()
+	prefix := path.Join(gc.option.cli.RootPath(), common.AnalyzeStatsPath) + "/"
+	// list dir first
+	keys, _, err := gc.option.cli.ListWithPrefix(ctx, prefix, false)
```

Comment:
This list may cost much. Considering partition key collection may create partition_num analyze task in one major compaction, we may have hundreds or thousands analyze file at one time even Milvus all work well. Just a comment, it is long term problem. Or we can change the path pattern to optimize a bit？

Reply:
It sounds reasonable, but it feels more appropriate to put collectionID/partitionID before taskID.